sale_order_import/wizard/sale_order_import.py

```diff
@@ -181,14 +182,20 @@ def _prepare_order(self, parsed_order, price_source):
                         parsed_order['order_ref'],
                         existing_orders[0].name,
                         existing_orders[0].state))
-        partner_change_res = soo.onchange_partner_id(partner.id)
-        assert 'value' in partner_change_res, 'Error in partner change'
+        values = {'partner_id': partner.id}
+        fake_so = soo.new(values)
+        fake_so.onchange_partner_id()
+        partner_change_res = dict(
+            (key, value.id if not isinstance(value, int) else value)
+            for key, value in dict(fake_so._cache).iteritems()
+        )
```

Comment:
There must be a prettier way to do this, the issue was that `dict(fake_so._cache)` returns the objects in a dict, whereas we need the id's.